src/api_client/api_resource.js

```diff
@@ -39,7 +39,15 @@ export default class APIResource {
 		const request = popsicle
 			.request(req)
 			.use(popsicle.plugins.parse(['json', 'urlencoded']))
-			.then(res => res.body);
+			.then(res => {
+				if (res.status >= 300) {
+					if (res.body && res.body.message) {
+						throw new Error(res.body.message);
+					}
+					throw new Error('An unknown error has occurred.');
```

Comment:
How about `An unknown error in the API response has occured.` Maybe also provide the statusCode so that users have some insight?

Reply:
I think the phrase `An unknown error has occurred.` is common, and since the user is doing the request function, he should know it is an API response error.
but i think it's good to have status code